### ISSUE TITLE
Javadoc fail on @link reference not found

### ISSUE THREAD

**Issue by ryanidev** [checkstyle/checkstyle#915]
Hello,

 I'm looking to have my build fail if a @link reference is not found. I've been searching for a couple days to see if there's an option, and for the life of me, I can't find it. I've been able to get the build to fail on just about every other warning under the sun, but this one doesn't fail the build.

Any help would be greatly appreciated.

Thanks,
Ryan

**Comment by romani** [comment]
It would help a lot if you can specify name of Check that cause violation.

Quick solution - deactivate a Check. 
Less quick solution is to suppress that violations by Filters http://checkstyle.sourceforge.net/config.html , there are bunch of them

JavaDoc Checks are buggy, I would recommend to switch them off or suppress as problem found. We will re-implement them at some period of time , probably this year.

**Comment by ryanidev** [comment]
There's no violation, per se. Everything is building fine. What I would like to happen, is if an @link reference is not found, it should fail the build. We're looking to fail the build if anything like this comes up. The original idea was to fail the build on ANY javadoc warnings, but there are a few we don't care about (not EVERY method will have javadoc at all, we're not concerned about missing @params, or @returns either).

So, I guess I didn't communicate properly. Sounds like you think I'm having things fail, it's the opposite, things are ok, but I WANT the build to fail, if javadoc reports that an @link reference is not found. This is for an open source project, and would like to make sure ALL javadoc @links work properly.

Thanks for the reply!

**Comment by romani** [comment]
Please state a small chunk of code that you have and code that you would like to have. Please make a link to your project and we will test our fix on your code.

**Comment by ryanidev** [comment]
Well, what I would think would make the most sense, is something in the checkstyle.xml file, something like this (the default being true)
```xml
<module name="JavadocMethod">
    <property name="allowBrokenLinkReferences" value="false" /> 
</module>
```

Currenlty, I don't have anything that checks for this. It's why I hooked up checkstyle in the first place. 

The project is here: https://github.com/iDevicesInc/SweetBlue

Currently, there are no broken @link references, so to test, just open any of the java files and add an extra character to one of the links, thats what I've been doing to test things out. Let me know if there's anything else I can provide to help out.

**Comment by romani** [comment]
that is not a simple feature request as it is the same as functionality of https://maven.apache.org/plugins/maven-linkcheck-plugin/ , example : http://checkstyle.sourceforge.net/linkcheck.html it has a lot of HTTP nuances , error codes, timeouts, ..... even for your sources there a lot of nuances  , look at:
```
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-linkcheck-plugin</artifactId>
```
 https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L1083

**Comment by mkordas** [comment]
@romani, I think this is about broken links to classes, not to external resources. We had the same issue. Neither PMD, FindBugs or Linkcheck reports them - but its capability of Java itself. They are reported as warnings during Javadoc building. I've repaired our violations in 734c516bd982b9046b0b84ff9d255fc570f54bc7 (part of #291).

**Comment by romani** [comment]
I am ok with local references, but that have to be reflected in option name and its future description.
We will think more about that issue , when we come to all Javadoc Checks reimplementation to use AST tree instead of regexp.

**Comment by ryanidev** [comment]
@mkordas is correct. I'm specifically talking about links to classes in this instance (though we'd probably also want to make sure http links work as well).

I suppose I'll have to go the hacky, messy way, and scrape the console output to see if that warning has come up, and fail if it does.

I honestly didn't think this would be such a giant pain. I essentially just want to treat those warnings as errors. This isn't a mission critical thing, just something to ensure that we don't push a release with broken class @links.

**Comment by romani** [comment]
we will focus on local links first.

remote links we might be covered as next step, it is also useful, but have a lot of nuances and I do not want to release unreliable implementation.

**Comment by ryanidev** [comment]
Makes sense. Local links are definitely more important anyways.

**Comment by romani** [comment]
Checkstyle limitation that during validation there is no other files, no class path. 
https://checkstyle.org/writingchecks.html#Limitations

We conceptually can not make cover validation of javadoc inlined tags to other classes or their methods.

Maybe be future we come back to support validations with classpath in mind, but it is not easy as forcing users to extra complicated configuration for checkstyle execution.

I am closing this issue for now
